### PR TITLE
Fix gamma irrep

### PR DESCRIPTION
**Pull request by AliGhorashiCMT**
Same as previous pull request, but hopefully it works this time

### CODE REVIEW

**Comment by AliGhorashiCMT** [comment]
Just saw the suggested changes on the other pull request- will do them here now

**Comment by AliGhorashiCMT** [comment]
Should be done now

**Comment by thchr** [comment]
See also my comment regarding the 3D case.

**Comment by AliGhorashiCMT** [comment]
I think the latest commit fixed that hopefully

File: src/read_utils.jl [hunk 1]
```diff
@@ -70,6 +70,33 @@ function _find_next_separable_band_grouping(
...
+
+end
+
+function fix_gamma_irrep(symeigsd::Dict{String, Vector{Vector{ComplexF64}}}, lg_gamma::LittleGroup{D},
+    polarization::Union{AbstractString, Nothing}=nothing) where D
+    symeigsd_fixed = symeigsd
+    signs = [det(rotation(g)) for g in lg_gamma]
+    if D == 2
+        if polarization == "tm"
+            symeigsd_fixed["Γ"][1] = ones(length(signs))
```
Comment (thchr) [inline src/read_utils.jl:87]:
```suggestion
            symeigsd_fixed["Γ"][1] .= 1
```

File: src/read_utils.jl [hunk 2]
```diff
@@ -70,6 +70,33 @@ function _find_next_separable_band_grouping(
...
+        elseif polarization == "te"
+            symeigsd_fixed["Γ"][1] = signs
```
Comment (thchr) [inline src/read_utils.jl:89]:
```suggestion
            symeigsd_fixed["Γ"][1] .= signs
```

File: src/read_utils.jl [hunk 3]
```diff
@@ -70,6 +70,33 @@ function _find_next_separable_band_grouping(
...
+        end
+    elseif D ==3
+        x2T = signs .* (2cospi.(2 ./ Crystalline.rotation_order.(lg_gamma)) .+ 1) .- 1
+        symeigsd_fixed[Γ][1] = x2T*0.65
+        symeigsd_fixed[Γ][2] = x2T*0.35
```
Comment (thchr) [inline src/read_utils.jl:94]:
```suggestion
        arbitrary_split = 0.3821
        symeigsd_fixed[Γ][1] .= x2T * arbitrary_split
        symeigsd_fixed[Γ][2] .= x2T * (1 - arbitrary_split)
```

**Comment by AliGhorashiCMT** [comment]
I committed all the suggestions- after merging, I'll check if all my code still works as intended and open another pull request for the bandsummary addition and length methods

**Comment by thchr** [comment]
I tidied a few things up in https://github.com/thchr/MPBUtils.jl/pull/13/commits/6e6171ea8ce1c5a473284112992b6f56c10d546a (most relevant bit: this call inevitably mutates the symmetry eigenvalues - so it should be postfixed by `!`, per convention).

If you could write a doc-string, we can merge this (preferably, can you test out the functionality _before_ we merge?).

**Comment by thchr** [comment]
Okay, I just went ahead and added a doc-string. I also changed the name to `fixup_gamma_symmetry` since we are not actually working with irreps.

If you could check that it is doing the job we want in 2D, that'd be great!

**Pull request merged**